src/ryu_faucet/org/onfsdn/faucet/gauge.py

```diff
@@ -96,6 +101,36 @@ def __init__(self, *args, **kwargs):
             self.watchers.setdefault(watcher.dp.dp_id, {})
             self.watchers[watcher.dp.dp_id][watcher.conf.type] = watcher
 
+        # Database specific config file read
+        self.db_enabled = False
+        self.flow_database = None
+        self.switch_database = None
+        self.conn = None
+        with open(self.db_config, 'r') as stream:
+            data = yaml.load(stream)
+            if data['database']:
+                self.db_enabled = True
+                self.conn_string = "driver={0};server={1};port={2};" \
+                                   "uid={3};pwd={4}".format(
+                    data['driver'], data['db_ip'], str(data['db_port']),
+                    str(data['db_username']), str(data['db_password'])
+                )
+                nsodbc = nsodbc_factory()
+                self.conn = nsodbc.connect(self.conn_string)
+                self.switch_database, exists = self.conn.create(
+                                                data['switches_doc'])
+                # Create database specific views for querying
+                if not exists:
+                    init_switch_db(self.switch_database)
+
+                self.flow_database, exists = self.conn.create(
+                                                data['flows_doc'])
+                # Create database specific views for querying
+                if not exists:
+                    init_flow_db(self.flow_database)
+                self.db_conf_data = data
+
+
```

Comment:
Reading the config should be done in config_parser.py, then the parsed config should be giving to a watcher, and any initialising of the database can be done when that watcher is initialised. If you need to have that happen only once, then let the watcher call some kind of function in nsodbc.py that ensures that.